.github/workflows/main.yml

```diff
@@ -15,6 +15,10 @@ on:
     branches:
       - '**'
 
+concurrency:
+  group: ${{ github.head_ref }}
+  cancel-in-progress: true
```

Comment:
do you want to cancel in-progress tests on the e.g. `master` branch (this will result in ❌ when skipped)? or just PR's? if the latter 
```suggestion
  cancel-in-progress: ${{ github.event_name == 'pull_request' }}
```

Reply:
I was reasonably sure from the docs that `${{ github.head_ref }}` would only be present during PRs, and an empty concurrent group would prevent this from taking effect. Did you find something to the contrary?

Reply:
ah, nope! just missed the `head_ref` only being present on PR's